### DISCUSSION TITLE
Device-specific long URLs option missing in Webclient 4.1.0

### DISCUSSION THREAD

**Discussion by boschkundendienst** [shlinkio/shlink-web-client#1207] [General]
### shlink-web-client version

4.1.0

### How do you use shlink-web-client

Docker image

### Current behavior

In previous version `<4.1.0` one could add "Device-specific long URLs". When using Shlink-Web-Client `4.1.0` this option is no longer visible in the UI even when switching to "Advanced Options".
Is this by design (feature removed) or a bug?
Also tested with the web client hosted at https://app.shlink.io/

### Expected behavior

When using "Advanced Options" Device-specific long URLs should be possible.

### Minimum steps to reproduce

```
docker run -p 8080:8080 ghcr.io/shlinkio/shlink-web-client:4.0.0 # works as expected
```

```
docker run -p 8080:8080 ghcr.io/shlinkio/shlink-web-client:4.1.0 # is missing Device-specific long URLs in UI
```

**Comment by acelaya** [comment]
Starting with shlink-web-client 4.1.0, the "Device-specific long URLs" section is dynamically displayed/hidden based on the version of the Shlink server you are using.

Shlink 4.0.0 replaced the device-specific redirects with the new [Dynamic rule-based redirects](https://github.com/shlinkio/shlink/discussions/1912) system, which covers the same functionality, but is more flexible.

If you are using Shlink >= 4.0, you now have to go to "Manage redirect rules", in the short URL dropdown, in order to configure redirects based on the device type.

![redirect-rules-menu](https://github.com/shlinkio/shlink-web-client/assets/2719332/f93e1e25-3c0c-4c49-aa43-939285812c04)

**Comment by boschkundendienst** [comment]
Thanks for clarifying. With the new implementation, there are now even more options available. The only thing I don't like is that you now have to create a short link first and then modify it afterwards. If you used to work with the advanced options, you now have to perform one more step.

Thanks for your help and this great tool!
